test/utility/misc.c

```diff
@@ -323,13 +325,94 @@ testutil_create_backup_directory(const char *home)
+ */
+void
+testutil_verify_src_backup(WT_CONNECTION *conn, const char *backup, const char *home)
+{
+    struct stat sb;
+    WT_CURSOR *cursor, *file_cursor;
+    WT_DECL_RET;
+    WT_SESSION *session;
+    uint64_t cmp_size, next_offset, offset, size, type;
+    int i, j, status;
+    char buf[1024], *filename, *id[WT_BLKINCR_MAX];
+    const char *idstr;
+
+    WT_CLEAR(buf);
+    testutil_check(conn->open_session(conn, NULL, NULL, &session));
+    testutil_check(session->open_cursor(session, "backup:query_id", NULL, buf, &cursor));
+    i = 0;
+    while ((ret = cursor->next(cursor)) == 0) {
+        testutil_check(cursor->get_key(cursor, &idstr));
+        id[i++] = dstrdup(idstr);
+    }
+    testutil_check(cursor->close(cursor));
+    testutil_assert(i <= WT_BLKINCR_MAX);
+    /* Go through each id and open a backup cursor on it to test incremental values. */
+    for (j = 0; j < i; ++j) {
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "incremental=(src_id=%s)", id[j]));
+        testutil_check(session->open_cursor(session, "backup:", NULL, buf, &cursor));
+        while ((ret = cursor->next(cursor)) == 0) {
+            testutil_check(cursor->get_key(cursor, &filename));
+            testutil_check(__wt_snprintf(buf, sizeof(buf), "incremental=(file=%s)", filename));
+            testutil_check(session->open_cursor(session, NULL, cursor, buf, &file_cursor));
+            next_offset = 0;
+            while ((ret = file_cursor->next(file_cursor)) == 0) {
+                testutil_check(file_cursor->get_key(file_cursor, &offset, &size, &type));
+                /* We only want to check ranges for files. So if it is a full file copy, ignore. */
+                if (type != WT_BACKUP_RANGE)
+                    break;
+                /*
+                 * If the offset indicates it is the expected next offset, then the blocks have
+                 * changed and we expect them to be different. If the offset is not the next
+                 * incremental block it was skipped and the blocks must be identical.
+                 */
+                testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/%s", backup, filename));
+                ret = stat(buf, &sb);
+                /*
+                 * The file may not exist in the backup directory. If the stat call doesn't succeed
+                 * skip this file. If we're skipping changed blocks go to the next one.
+                 */
+                if (ret != 0)
+                    break;
+                /*
+                 * If the block is changed we cannot check it (for differences, for example). The
+                 * source id may be older and we've already copied the block, or not, so we don't
+                 * know if it should be different or not. But if a block is indicated as unchanged
+                 * then we know for sure it better match.
+                 */
+                if (offset > next_offset) {
```

Comment:
I find the variable name, `next_offset` confusing here (and a couple lines below). It sounds like is should be *after* `offset`. But in fact it is earlier than `offset`. 

Maybe `prev_offset` or `last_offset`?

Reply:
Fixed.